platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -341,6 +355,71 @@ MGL_EXPORT
  */
 @property (nonatomic, readonly) unsigned long long countOfBytesCompleted;
 
+
+#pragma mark - Managing Ambient Cache
+
+/**
+ Sets the maximum ambient cache size in megabytes. The default maximum cache
+ size is 50 MB. To disable ambient caching, set the maximum ambient cache size
+ to 0. Setting the maximum ambient cache size does not impact the maximum size
+ of offline packs.
+ 
+ If you intend to use offline packs, this may limit the amount of space available
+ for ambient caching, with no impact on offline packs. If you set the maximum
+ ambient cache size to 30 MB and offline packs use 20 MB, there may only be 10 MB
+ reserved for the ambient cache.
+ 
+ This method should be called before the map and map style have been loaded.
+ 
+ This method is potentially expensive, as the ambient cache will trim cached data
+ to prevent the database from being larger than the specified amount.
+ 
+ @param cacheSize The maximum size in bytes for the ambient cache.
+ @param completion The completion handler to call once the maximum ambient cache size
+ has been set. This handler is executed synchronously on the main queue.
+ */
+
+- (void)setMaximumAmbientCacheSize:(NSUInteger)cacheSize withCompletionHandler:(void (^)(NSError *_Nullable error))completion;
+
+/**
+ Checks that the tiles in the ambient cache match those from the server. Local
+ tiles that do not match the latest version on the server are updated.
+ 
+ This is more efficient than cleaning the cache because valid local tiles will
+ not be downloaded again.
+ 
+ Resources shared with offline packs will not be affected by this method.
+ 
+ @param completion The completion handler to call once the ambient cache has
+ been revalidated. This handler is executed asynchronously on the main queue.
+ */
+
+- (void)invalidateAmbientCacheWithCompletionHandler:(void (^)(NSError *_Nullable error))completion;
+
+/**
+ Erase resources from the ambient cache.
+ 
+ Resources that are shared with offline regions will not be affected by
+ this method.
+ 
+ @param completion The completion handler to call once the ambient cache has been
+ cleared. This handler is executed asynchronously on the main queue.
+ */
+
+- (void)clearAmbientCacheWithCompletionHandler:(void (^)(NSError *_Nullable error))completion;
```

Comment:
I believe it is because the database is deleted, then reinitialized, while resources related to the ambient cache are deleted/cleared from the database without any reinitialization.